src/researchhub_comment/views/rh_comment_thread_view_mixin.py

```diff
@@ -0,0 +1,124 @@
+from rest_framework.response import Response
+from rest_framework import status
+from rest_framework.decorators import action
+from rest_framework.permissions import (
+    AllowAny,
+    IsAuthenticated,
+)
+
+from researchhub_comment.related_models.rh_comment_model import RhCommentModel
+from researchhub_comment.related_models.rh_comment_thread_model import (
+    RhCommentThreadModel,
+)
+
+
+class RhCommentThreadViewMixin:
+    @action(detail=True, methods=["POST"], permission_classes=[IsAuthenticated])
+    def create_rh_comment(self, request, pk=None):
+        try:
+            rh_thread = self._retrieve_or_create_thread_from_request_data(request)
+            _rh_comment = RhCommentModel.create_from_data(request.data, rh_thread)
+            rh_thread.refresh_from_db()  # object update from fresh db values
+            return Response(self.get_serializer(instance=rh_thread).data, status=200)
+        except Exception as error:
+            return Response(
+                f"Failed - create_rh_comment: {error}",
+                status=status.HTTP_400_BAD_REQUEST,
+            )
+
+    @action(detail=True, methods=["POST"], permission_classes=[AllowAny])
+    def get_comment_threads(self, request, pk=None):
+        raise NotImplementedError("Needs to be implemented")
+
+    def _retrieve_or_create_thread_from_request_data(self, request):
+        request_data = request.data
+        try:
+            thread_id = request_data.get("thread_id") or None
+            if thread_id is not None:
+                return RhCommentThreadModel.objects.get(id=int(thread_id))
+            else:
+                existing_thread = self._get_existing_thread_from_request(request)
+                if existing_thread is not None:
+                    return existing_thread
+                else:
+                    valid_thread_target_model = (
+                        RhCommentThreadModel.get_valid_thread_target_model(
+                            self._resolve_target_model_name(request)
+                        )
+                    )
+                    thread_target_instance = valid_thread_target_model.objects.get(
+                        id=self._resolve_target_model_instance_id(request)
+                    )
+                    return thread_target_instance.rh_threads.create(
+                        thread_type=request_data.get("thread_type"),
+                        thread_reference=request_data.get("thread_reference"),
```

Comment:
What does this do exactly?

Reply:
It grabs the target model class instance that is meant to create generic relation to Thread. (For example Paper class instance). Then creating a thread accordingly